### PR TITLE
Configure Vagrant for NFS shares and document it.

### PR DESCRIPTION
**Pull request by rbarlow**
This commit adds documentation of our new Vagrant setup, as well as
configuring it to use NFS to share the code rather than rsync which was
very slow.

### CODE REVIEW

File: docs/dev-guide/contributing/dev_setup.rst [hunk 1]
```diff
@@ -5,12 +5,90 @@ Developer Setup
...
+   configures your NFS exports, and everything should work. Another possible failure point is during
+   provisioning when mongod is building its initial files. This sometimes takes longer than systemd
+   allows and can fail. If that happens, simple run ``vagrant provision``. We will run two vagrant
+   reloads in a row. The first is a workaround for the mentioned bug, and the second allows the
+   machine to reboot after provisioning.::
+
+   $ cd pulp
+   $ vagrant up  # You will experience an error starting nfs and rpcbind here
+   $ vagrant reload  # mongod may fail when this runs. vagrant provision will fix it.
+   $ vagrant reload  # Reboot the machine at the end to apply kernel updates, etc.
```
Comment (beav) [inline docs/dev-guide/contributing/dev_setup.rst]: I get stuck on this step, even after running `vagrant reload` a few times:

```
==> default: Mounting NFS shared folders...
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

mount -o 'vers=4' 192.168.121.1:'/path/to/my/code/dir' /home/vagrant/devel

Stdout from the command:



Stderr from the command:

mount.nfs: access denied by server while mounting 192.168.121.1:/path/to/my/code/dir
```

Comment (rbarlow) [inline docs/dev-guide/contributing/dev_setup.rst]: On 04/23/2015 07:48 PM, Chris Duryee wrote:
> |mount.nfs: access denied by server while mounting 192.168.121.1:/path/to/my/code/dir|

This is the error message I got before I learned that I needed to grant
r-x access to the nfsnobody user to my home dir. You might need to give
that user access to one of the parent dirs (i.e., if /path/to is 700,
nfsnobody will need a facl to /path/to in order to get to the full
path). Do you think that is the problem, or something else?

Another possibility is that it's selinux related. I didn't need to
adjust any selinux rules in order to serve /home/rbarlow/devel, but
perhaps if you are trying with a different path you will need special
SELinux contexts?

-- 
Randy Barlow

File: docs/dev-guide/contributing/dev_setup.rst [hunk 2]
```diff
@@ -5,12 +5,90 @@ Developer Setup
...
+
+#. Install NFS. NFS will be used to share your code directory with the deployed virtual machine::
+   
+   $ sudo yum install nfs-utils
+
+#. You will need to grant the nfsnobody user rx access to the folder that you check out your code
+   under. Many developers check out code into $HOME/devel or similar. In Fedora, $HOME typically
+   does not allow such access to other users. If your code is in $HOME, you will need to::
+   
+   $ setfacl -m user:nfsnobody:r-x $HOME  # Season to taste, as per above
```
Comment (beav) [inline docs/dev-guide/contributing/dev_setup.rst]: If your code lives in a dir under $HOME, you will need to setfacl on both $HOME and the dir underneath.

**Comment by beav** [comment]
I was able to do a checkout and run the unit tests from within a Vagrant machine. Very cool!

File: docs/dev-guide/contributing/dev_setup.rst [hunk 3]
```diff
@@ -5,12 +5,90 @@ Developer Setup
...
+   $ git clone git@github.com:pulp/pulp_puppet
+   $ git clone git@github.com:pulp/pulp_ostree
+
+#. Next, cd into the pulp directory, and begin provisioning your Vagrant environment. Due to what
+   seems to be a `Vagrant bug <https://github.com/mitchellh/vagrant/issues/5321>`_, the first run
+   will fail due to Vagrant attempting to control NFS through init scripts rather than through
+   systemd. There is an easy workaround, as you can tell Vagrant to reload the box after it
+   configures your NFS exports, and everything should work. Another possible failure point is during
+   provisioning when mongod is building its initial files. This sometimes takes longer than systemd
+   allows and can fail. If that happens, simple run ``vagrant provision``. We will run two vagrant
```
Comment (mhrivnak) [inline docs/dev-guide/contributing/dev_setup.rst]: s/simple/simply

File: docs/dev-guide/contributing/dev_setup.rst [hunk 4]
```diff
@@ -5,12 +5,90 @@ Developer Setup
...
+   machine to reboot after provisioning.::
+
+   $ cd pulp
+   $ vagrant up  # You will experience an error starting nfs and rpcbind here
+   $ vagrant reload  # mongod may fail when this runs. vagrant provision will fix it.
+   $ vagrant reload  # Reboot the machine at the end to apply kernel updates, etc.
+
+Once you have followed the steps above, you should have a running deployed Pulp development machine.
+You can ssh into the environment with ``vagrant ssh``. All of the code is mounted in
+/home/vagrant/devel. You can simply type ``workon pulp`` to work on the pulp checkout, or
```
Comment (mhrivnak) [inline docs/dev-guide/contributing/dev_setup.rst]: For those unfamiliar, can you elaborate on what this does?

File: docs/dev-guide/contributing/dev_setup.rst [hunk 5]
```diff
@@ -5,12 +5,90 @@ Developer Setup
...
+
+   $ cd $HOME/devel  # Season to taste
+   $ git clone git@github.com:pulp/pulp.git && git clone git@github.com:pulp/pulp_rpm.git
+
+#. Optionally, check out some of our other plugins as well::
+
+   $ git clone git@github.com:pulp/pulp_docker
+   $ git clone git@github.com:pulp/pulp_python
+   $ git clone git@github.com:pulp/pulp_puppet
+   $ git clone git@github.com:pulp/pulp_ostree
```
Comment (rbarlow) [inline docs/dev-guide/contributing/dev_setup.rst]: Perhaps add a note that these should all be checked out to compatible branches, or perhaps just say master.

File: docs/dev-guide/contributing/dev_setup.rst (hunk 1 above)
Comment (rbarlow) [inline docs/dev-guide/contributing/dev_setup.rst]: Documented.

File: docs/dev-guide/contributing/dev_setup.rst (hunk 2 above)
Comment (rbarlow) [inline docs/dev-guide/contributing/dev_setup.rst]: Documented.

File: docs/dev-guide/contributing/dev_setup.rst (hunk 3 above)
Comment (rbarlow) [inline docs/dev-guide/contributing/dev_setup.rst]: Fixed.

File: docs/dev-guide/contributing/dev_setup.rst (hunk 4 above)
Comment (rbarlow) [inline docs/dev-guide/contributing/dev_setup.rst]: Fixed!

File: docs/dev-guide/contributing/dev_setup.rst (hunk 5 above)
Comment (rbarlow) [inline docs/dev-guide/contributing/dev_setup.rst]: Fixed.

**Comment by rbarlow** [comment]
ok test

**Comment by rbarlow** [comment]
ok test

**Comment by rbarlow** [comment]
@beav @mhrivnak @jortel Can I get an LGTM from one of you?

**Comment by beav** [comment]
LGTM, great work :cake:

**Pull request merged**